tests/vmcheck/multitest.py

```diff
@@ -0,0 +1,104 @@
+#!/bin/env python
+
+import os
+import sys
+import glob
+import subprocess
+
+
+def main():
+
+    failed = False
+    hosts = []
+    for host in sys.argv[1:]:
+        hosts.append(Host(host))
+
+    for test in glob.iglob(os.path.join(sys.path[0], "test-*.sh")):
+        host = wait_for_next_available_host(hosts)
+        rc = host.flush()
+        failed = failed or rc != 0
+        host.dispatch(test)
+
+    for host in hosts:
+        rc = host.flush()
+        failed = failed or rc != 0
+
+    # fetch the journal from all the hosts which had a failure
+    fetcher = os.path.join(sys.path[0], "fetch-journal.sh")
+    for host in hosts:
+        if host.saw_fail:
+            fetcher_env = dict(os.environ)
+            fetcher_env.update({'VM': host.hostname,
+                                'JOURNAL_LOG':
+                                "vmcheck/%s.journal.log" % host.hostname})
+            subprocess.check_call([fetcher], env=fetcher_env)
+
+    return 1 if failed else 0
+
+
+def wait_for_next_available_host(hosts):
+    while True:
+        for host in hosts:
+            if host.is_done():
+                return host
+        os.wait()
```

Comment:
This is where one really wants a main event loop (+ state machine).  The main options here are the GLib one, or the Python 3.5-ish one.  The latter would require jumping through some hoops for our CentOS builds, but manageable.

But let's not respin this unless we have to, as I see this as a short term fix.